### PR TITLE
Add update event form

### PR DESCRIPTION
**Pull request by cmaija**

### CODE REVIEW

**Comment by cmaija** [comment]
hello yes it me finally done and ready to die

File: parkhangs-expressapp/models/EventModel.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ const Event = new Schema(
         eventDateTime: {type: String, required: true},
         createdDateTime: {type: String, required: true},
         creatorName: {type: String, required: true},
-        creatorId: {type: Number, required: true}
+        creatorId: {type: Number, required: false}
```
Comment (gordonzhang17) [inline parkhangs-expressapp/models/EventModel.js:11]: why is this changed?

File: parkhangs-expressapp/controllers/index.js [hunk 2]
```diff
@@ -27,5 +30,8 @@ const getParks = async (req, res) => {
 };
 
 module.exports = {
-    getParks
+    getParks,
+    getParks,
```
Comment (gordonzhang17) [inline parkhangs-expressapp/controllers/index.js:34]: duplicate

File: parkhangs-webapp/src/components/AddEventForm.js [hunk 3]
```diff
@@ -51,35 +81,41 @@ class AddUpdateEventForm extends React.Component {
...
+        if (!this.props.eventId) {
            this.props.addOneEvent(this.props.parkId, detail, eventDateTime)
         }
-        // else {
-        //    this.props.updateEvent(this.props.parkId, event)
-        // }
+
+        else {
+            this.props.updateEvent(this.props.eventId, this.props.parkId, detail, eventTimestamp)
+            this.props.updatedEvent(this.props.eventId, detail, eventTimestamp)
```
Comment (lydia-zheng) [inline parkhangs-webapp/src/components/AddEventForm.js:99]: Is this duplicate too?

File: parkhangs-expressapp/controllers/index.js (hunk 2 above)
Comment (cmaija) [reply parkhangs-expressapp/controllers/index.js:34]: thanks yo

File: parkhangs-expressapp/models/EventModel.js (hunk 1 above)
Comment (cmaija) [reply parkhangs-expressapp/models/EventModel.js:11]: because I wasn't allowed to update an event without providing a creator id. I can change it back and add a fake one in my controller.

File: parkhangs-webapp/src/components/AddEventForm.js (hunk 3 above)
Comment (cmaija) [reply parkhangs-webapp/src/components/AddEventForm.js:99]: no. could have named it better but it's so the parent component reflects the change.

File: parkhangs-webapp/src/features/parks/parksSlice.js [hunk 4]
```diff
@@ -92,6 +90,61 @@ const parksSliceReducers = {
...
+
+          prepare (error) {
+              return {
+                  payload: { error }
+              }
+          }
+      },
 
       selectPark: {
         reducer(state, action) {
```
Comment (gordonzhang17) [inline parkhangs-webapp/src/features/parks/parksSlice.js:150]: we can remove this

File: parkhangs-webapp/src/components/AddEventForm.js (hunk 3 above)
Comment (cmaija) [reply parkhangs-webapp/src/components/AddEventForm.js]: changed the name for clarity

File: parkhangs-webapp/src/features/parks/parksSlice.js (hunk 4 above)
Comment (cmaija) [reply parkhangs-webapp/src/features/parks/parksSlice.js:150]: :+1:
